app/main.py

```diff
@@ -1,3 +1,50 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        raise TypeError("Unsupported operand type(s)")
+
+    def __iadd__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported operand type(s)")
+        return self
+
+    def __mul__(self, other: float) -> "Distance":
```

Comment:
The type hint for the `other` parameter should include `int` as well, since the method checks for both `int` and `float`. Change `other: float` to `other: int | float`.